File: four_rings/lib/four_rings/token.ex
defmodule FourRings.Token do
  @moduledoc """
  Work token that circulates through a ring.
  """

  defstruct [
    :token_id,
    :ring_id,
    :orig_input,
    :current_val,
    :remaining_hops,
    :started_at
  ]
end

File: four_rings_java/src/fourrings/Token.java
package fourrings;

public class Token {

    public long tokenId;
    public String ringId;
    public long origInput;
    public long currentVal;
    public long remainingHops;
    public long startTime;

    public Token(long tokenId, String ringId, long input, long hops) {
        this.tokenId = tokenId;
        this.ringId = ringId;
        this.origInput = input;
        this.currentVal = input;
        this.remainingHops = hops;
        this.startTime = System.nanoTime();
    }
}